index.rst

```diff
@@ -1,61 +1,311 @@
+.. _original uml file from v0.4.5: https://github.com/lsst-ts/ts_MTAOS/blob/v0.4.5/doc/uml/mtaosClass.uml
+
+Nevertheless, the `Model class`_ itself will require substantial updates to incorporate the new functionality mentioned :ref:`above <Data-access-changes>` and the conversion of `WEP`_ into a gen3 pipeline task.
+
+Updates to the Model class
+--------------------------
+
+The `Model class`_ has two main attributes it uses to perform the wavefront estimation (`WEP`_) and to convert the output to optical corrections (`OFC`_).
+
+The ``wep`` attribute (and its uses) is probably the one that will require the most work.
+First, we have to keep in mind that the ``wep`` will have two modes of operation; local and remote.
+
+The local mode will be used for the closed loop, e.g., analysis of the corner wavefront sensors for the main camera.
+Whereas, remote mode will be used (mostly) for full focal plane analysis (specially for ``LSSTCam``).
+
+Although ComCam data could be analyzed in local mode, it is probably useful to allow users to select if they want ComCam to be processes locally or remotely.
+This would allow us to test both use cases early on.
+
+With respect to the ``ofc`` attribute, there is little that needs to be done.
+The attribute should remain in the ``Model`` class and should continue to be used as is.
+Most of what this class does, once configured with the correct instrument setup, is to receive a list of wavefront errors and convert them to corrections for each of the optical components.
+Some updates to the ``Model`` class will have to be implemented to support the :ref:`user-provided-optical-aberration` feature, but should be straightforward to implement.
 
-   **This technote is not yet published.**
+Fortunately, the use of ``wep`` and ``ofc`` is well abstracted in the ``Model`` class, being restricted to some specific methods that are called by the CSC.
 
-   With integration of the MTAOS with the Main Telescope components approaching we have identified some design decisions made in the past that needs revisiting. This technote explores some design considerations for improvements to the MTAOS operation.
+.. _OFC: https://github.com/lsst-ts/ts_ofc
 
-.. Add content here.
-.. Do not include the document title (it's automatically added from metadata.yaml).
+Miscellaneous
+=============
 
-.. .. rubric:: References
+These are a couple of improvements we should probably implement.
 
-.. Make in-text citations with: :cite:`bibkey`.
+Unify configuration schema
+--------------------------
 
-.. .. bibliography:: local.bib lsstbib/books.bib lsstbib/lsst.bib lsstbib/lsst-dm.bib lsstbib/refs.bib lsstbib/refs_ads.bib
-..    :style: lsst_aa
+The MTAOS currently has two configuration schemas; one for the CSC and a separate one called ``telescopedof.yaml``, which contains the configuration for the the OFC class.
+We should probably merge this second configuration file into the CSC configuration.
```

Comment:
The file of `telescopedof.yaml` was added by Petr at DM-24610. And I prefer two have multiple configuration files contains the parameters what to overwrite from `ts_MTAOS` to `ts_wep` and `ts_ofc`. This is easier to do the maintenance. For example, I use the sublime text and vscode as IDEs. You could see such two IDEs will separate different configuration parameters to different levels and files for the maintenance. It might be hard to read a configuration file that has more than 100 settings.

Reply:
> The file of `telescopedof.yaml` was added by Petr at DM-24610. And I prefer to have multiple configuration files contains the parameters what to overwrite from `ts_MTAOS` to `ts_wep` and `ts_ofc`. This is easier to do the maintenance. For example, I use the sublime text and vscode as IDEs. You could see such two IDEs will separate different configuration parameters to different levels and files for the maintenance. It might be hard to read a configuration file that has more than 100 settings.